.github/actions/setup-tebako/action.yml

```diff
@@ -0,0 +1,171 @@
+          apk --no-cache --upgrade add build-base cmake git bash   \
+          autoconf boost-static boost-dev flex-dev bison make      \
+          binutils-dev libevent-dev acl-dev sed python3 pkgconfig  \
+          lz4-dev openssl-dev zlib-dev xz ninja zip unzip curl     \
+          libunwind-dev libdwarf-dev gflags-dev elfutils-dev       \
+          libevent-static openssl-libs-static lz4-static xz-dev    \
+          zlib-static libunwind-static acl-static tar libffi-dev   \
+          gdbm-dev yaml-dev yaml-static ncurses-dev ncurses-static \
+          readline-dev readline-static p7zip ruby-dev gcompat      \
+          gettext-dev gperf brotli-dev brotli-static jemalloc-dev  \
+          fmt-dev xz-static clang
+          if [ "$CC" = "clang" ]; then
+            apk --no-cache --upgrade add clang
+          fi
+          if [ "${{ inputs.cache }}" != "off" ]; then
+            gem install bundler
+          fi
+        else
+          sudo apt-get -y update
+          sudo apt-get -y install                                                   \
+          binutils-dev libevent-dev acl-dev libfmt-dev libjemalloc-dev              \
+          libdouble-conversion-dev libiberty-dev liblz4-dev liblzma-dev libssl-dev  \
+          libboost-filesystem-dev libboost-program-options-dev libboost-system-dev  \
+          libboost-iostreams-dev  libboost-date-time-dev libboost-context-dev       \
+          libboost-regex-dev libboost-thread-dev libbrotli-dev libunwind-dev        \
+          libdwarf-dev libelf-dev libgoogle-glog-dev libffi-dev libgdbm-dev         \
+          libyaml-dev libncurses-dev libreadline-dev libutfcpp-dev
+        fi
+
+    - name: Select XCode
+      if: runner.os == 'macOS'
+      uses: maxim-lobanov/setup-xcode@v1
+      with:
+        xcode-version: ${{ env.XCODE_VERSION }}
+
+    - name: Install brew packages
+      if: runner.os == 'macOS'
+      shell: bash
+      run: |
+        brew update
+        # Already installed: openssl@1.1, libevent, libsodium, lz4, xz, zlib, gdbm, ncurses, pkg-config, libyaml
+        brew install bison flex binutils libffi double-conversion boost jemalloc fmt glog gnu-sed bash
+        echo "$(brew --prefix bison)/bin" >> $GITHUB_PATH
+
+    - name: Install Ruby
+      if: inputs.ruby_version != 'skip' && env.ALPINE_VER == ''
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: ${{ inputs.ruby_version != 'env' && inputs.ruby_version || env.RUBY_VER }}
+        bundler: ${{ inputs.bundler_version }}
+        bundler-cache: false
+
+    - name: Install tebako
+      shell: bash
+      run: gem install tebako
+
+    - name: Create deps folder
+      if: inputs.cache != 'off'
+      shell: bash
+      run: mkdir -p ${{ inputs.cache_path }}
+
+    - name: Process cache
+      if: inputs.cache != 'off'
+      uses: actions/cache@v4
+      with:
+        path: ${{ inputs.cache_path }}
+        key: ${{ runner.os }}-${{ env.CC }}-v${{ inputs.cache_ver }}
```

Comment:
This key means the same cache for Ubuntu and Alpine. That does not work.
Please see the comment above also

Reply:
`runner.os` replaced with `env.ADVANCED_OS`